firewall_podnet.py

```diff
@@ -184,8 +184,8 @@ def build(
     forward_rules = deque()
     for rule in sorted(firewall_rules, key=lambda fw: fw['order']):
         # sort traffic direction ie inbound, outbound and forward
-        iiface = rule['iiface'] if rule['iiface'] not in [None, '', 'none'] else None
-        oiface = rule['oiface'] if rule['oiface'] not in [None, '', 'none'] else None
+        iiface = rule['iiface'] if rule['iiface'] not in [None, '', 'none', 'any'] else None
+        oiface = rule['oiface'] if rule['oiface'] not in [None, '', 'none', 'any'] else None
```

Comment:
That's a good catch, corrected with the right way.
FYI, when in a rule if `iiface` is not mentioned means `iiface=any` similarly `oiface`, `saddr`, `daddr`, `dport` etc.

Primitive expects both can't be None at the same time because we can't tell the rule is inbound/outbound/ forward.

Reply:
So to your question, when the rule has both `iiface` and `oiface` are None, Primitive throws an error : https://github.com/CloudCIX/primitives/blob/1b1beaf4dafeb8f2783be6d5102475dc5fd22e89/controllers/firewall_podnet.py#L129

Reply:
With what was previously here before you fixed, this validation function should also have been extended to ensure both `iiface` and `oiface`  were not `any`.

Reply:
This was the change: https://github.com/CloudCIX/primitives/blob/1b1dc17ca1db8a07455d58329e318ff0e129711e/firewall_podnet.py#L26

As per the nftable rule, it is valid to have both `iiface` and `oiface` are not mentioned(which means iiface=any and oiface=any). 

Since it is Primitives thing to find out where to place the rule(inbound, outbound, farwarding), primitive uses `iiface` and `oiface` to find out the place. ie :
1. Inbound : for `iiface` except None it can take any interface or all interfaces and `oiface` should be None
2. Outbound: for `oiface` except None it can take any interface or all interfaces and `iiface` should be None
3. Forwarding: for `iiface` except None it can take any interface or all interfaces and `oiface` except None it can take any interface or all interfaces